### PR TITLE
Feature - Updated readme to provide detailed instructions

### PR DESCRIPTION
**Pull request by AryanSankholkar**
@iamleonie 

Hi, I have completed your readme for this project as required in the task.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,62 +1,266 @@
-# Dog Search 
-
-This project's origin is [here](https://github.com/weaviate/weaviate-examples/tree/main/nearest-neighbor-dog-search).
```
Comment (iamleonie) [inline readme.md:3]: Please keep this line

File: readme.md [hunk 2]
```diff
@@ -1,62 +1,266 @@
...
+### Running the Application
+
+To run your Flask application, use the following command:
+
 ```bash
-python query.py
+python flask-app/application.py
 ```
 
-## Dataset license
```
Comment (iamleonie) [inline readme.md:61]: Please keep the dataset license

**Review by iamleonie — CHANGES_REQUESTED** [review]
Hi @AryanSankholkar,
Thank you for your contribution. As our demos follow a simple template for the README.md file it would be great if you could keep your changes within the defined structure:
# [Title]
## Description
(description + demo video)
## Used Technologies
(Used technology stack + used Weaviate modules)
## Prerequisites

# Setup instructions

# Usage instructions

# Dataset license

Thank you so much.

**Comment by AryanSankholkar** [comment]
@iamleonie I have made the required changes in the readme file....Please have a look!

File: readme.md [hunk 3]
```diff
@@ -1,62 +1,70 @@
-# Dog Search 
+# Image Search Application with Weaviate
 
 This project's origin is [here](https://github.com/weaviate/weaviate-examples/tree/main/nearest-neighbor-dog-search).
 
 ## Description
-In this demo, we will use Weaviate to create an image-search application. We'll develop a dog image-search application that lets users upload a photo of a dog and receive a list of dog breeds that are the most similar.
+This guide demonstrates how to build an Image Search Application with Weaviate. The application allows users to upload images of dogs and find the most similar dog breed. While we focus on dogs, you can adapt this approach for various image recognition and product search applications.
 
-(TODO: Add demo video)
```
Comment (iamleonie) [inline readme.md:8]: Let's keep this line as well, so future contributors know that there is a TO DO here.

File: readme.md [hunk 4]
```diff
@@ -1,62 +1,70 @@
...
+   python weaviate-test.py
+   ```
+   You should see an output similar to: `{"classes": []}`.
+
+## Usage Instructions
+### Image Vectorization
+1. Add your images to the `flask-app/static/img` folder.
+2. Execute `images-to-base64.py` to convert the images to base64 encoding:
    ```bash
-   python -m venv .venv
```
Comment (iamleonie) [inline readme.md:24]: Please keep the setup of the virtual environment in the setup instructions

File: readme.md [hunk 5]
```diff
@@ -1,62 +1,70 @@
...
-1. Install the Python packages.
+### Weaviate Database
+1. Define the schema in `create-schema.py` to structure the data.
+2. Add the schema to Weaviate:
+   ```bash
+   python create-schema.py
+   ```
+3. Upload data objects in batches using `upload-data-objects.py`:
    ```bash
-    pip install -r requirements.txt
```
Comment (iamleonie) [inline readme.md:30]: Please also keep the installation of required packages in the setup instructions

File: readme.md [hunk 6]
```diff
@@ -1,62 +1,70 @@
...
-    python upload-data-objects.py
-    ```
-## Usage instructions
-### Run the application
-Run the Python Flask application and go to http://localhost:5001
-```bash
-python flask-app/application.py 
-```
-
-### Run the query
```
Comment (iamleonie) [inline readme.md:55]: Please also keep the step to run the query.py file

**Review by iamleonie — CHANGES_REQUESTED** [review]
Hey @AryanSankholkar, Thank you for implementing the requested changes. After the second review, I have added a few new comments regarding sections, where you were overwriting some important parts of the README.md file. 

What I usually recommend is to use a file comparison tool (e.g., native in VS Code, or Beyond Compare) to double-check that the changes you make are adding value and not removing important parts.

While we appreciate every contribution, it is important that they keep our projects in good health. Please let me know if you have any questions.

**Comment by AryanSankholkar** [comment]
have also made the required changes...Please have a look!!

File: readme.md (hunk 1 above)
Comment (iamleonie) [reply readme.md:3]: Hey, this got lost in your last commit again.

File: readme.md [hunk 7]
```diff
@@ -1,62 +1,87 @@
...
+
+- Weaviate
+- Vector DB
+- Python Programming Language (v3.10+)
+- Weaviate Python Client Library
+- Flask (For WebApp interface development)
+- Docker Desktop
+
+**Weaviate Modules:**
+- [List the Weaviate modules used]
```
Comment (iamleonie) [inline readme.md]: It looks like the module from your previous commit got lost?

**Review by iamleonie — CHANGES_REQUESTED** [review]
Hey, 
thanks for the updates again. 
It looks like a lot of the contributions from your previous commit got lost... (e.g., the description, Weaviate modules, explanation of what the images-to-base64.py does, running the test file), also the first line ("The project's origin is here...) was removed again. Could you please review and double-check your changes?

**Comment by AryanSankholkar** [comment]
Hi @iamleonie ,

I have once more updated the readme and this time with the correct changes. I am truly sorry for having to bother you again and again

I have added these parts back in the readme

> # Dog Search 
> 
> This project's origin is [here](https://github.com/weaviate/weaviate-examples/tree/main/nearest-neighbor-dog-search).

> **Weaviate Modules:**
> 
> [Image Vectorization](https://github.com/AryanSankholkar/DEMO-image-search-dogs/tree/feature/newfeature#image-vectorization)
> In this demonstration, we will work with a dataset of dog images. The current dataset includes ten images of different dog breeds. However, you can easily replace these images with any other dataset to suit your specific use case. To do this, add your images to the flask-app/static/img folder and execute the images-to-base64.py and upload-data-objects.py files.
> 
> For this use case, it's essential that image vectorization captures information about the dogs, such as breed, size, and color. Weaviate's img2vec-neural module is designed to solve this problem. It vectorizes each image, enabling semantic similarity-based searches.
> 
> [Img2vec-neural Module](https://github.com/AryanSankholkar/DEMO-image-search-dogs/tree/feature/newfeature#img2vec-neural-module)
> Weaviate's img2vec-neural module is a flexible vectorizer that converts images into meaningful vectors. It supports the ResNet-50 model, a Convolutional Neural Network (CNN) trained on the ImageNet database, which contains over 10 million images and 20,000 classes.

**Review by iamleonie — APPROVED** [review]
Thank you for your hard work on this. I will merge this now.

**Comment by iamleonie** [comment]
Related to https://github.com/weaviate-tutorials/DEMO-image-search-dogs/issues/1

**Comment by iamleonie** [comment]
Hey @AryanSankholkar,
Thank you for your contribution during Hacktoberfest 2023.
We'd like to send you some SWAG as a little thank you.
Would it be possible for you to reach out via our [Weaviate Slack](https://weaviate.io/slack) to discuss details?
Otherwise, please let me know how I can reach you best (e.g., LinkedIn, Twitter)
Thanks again and best regards
Leonie

**Comment by AryanSankholkar** [comment]
Thank you for letting me contribute towards Hacktoberfest 2023.You can reach me out through linkedin.My linkedin profle is Aryan Sankholkar.I have also sent you a follow for your ease.

**Pull request merged**